app/assets/javascripts/hubstats/application.js

```diff
@@ -14,45 +14,83 @@
 
 $(document).ready( function() {
+  defaultDateRange()
   setDateRange()
 });
 
+function defaultDateRange() {
+  $('.input-daterange').datepicker({
+    "todayHighlight": true,
+    "endDate": "Today",
+    "todayBtn": true
+  });
+};
+
 function setDateRange() {
-  var index = readCookie("hubstats_index") || 2;
-  var timer = document.getElementById("time-select");
+  if ((readCookie("hubstats_index") === "null~~null") || (readCookie("hubstats_index") === null)) {
+    var index = getDefaultDateRange();
+  } else {
+    var index = readCookie("hubstats_index")
+  }
 
-  timer.selectedIndex = index;
+  var timer = document.getElementById("submitDateRange");
+  dates = index.split("~~")
+  $('.input-daterange').find('[name="start"]').datepicker('update', new Date(dates[0]));
+  $('.input-daterange').find('[name="end"]').datepicker('update', new Date(dates[1]));
 
-  timer.onchange = function() {
-    createCookie("hubstats_index",this.selectedIndex,1);
+  timer.onclick = function() {
+    start_date = $('.input-daterange').find('[name="start"]').datepicker('getDate');
+    end_date = $('.input-daterange').find('[name="end"]').datepicker('getDate');
+    createCookie("hubstats_index", start_date, end_date, 1);
     window.location.reload();
   };
 };
 
-function createCookie(name,value,days) {
-    if (days) {
-        var date = new Date();
-        date.setTime(date.getTime()+(days*24*60*60*1000));
-        var expires = "; expires="+date.toGMTString();
-    }
-    else var expires = "";
-    document.cookie = name+"="+value+expires+"; path=/";
+function createCookie(name,value1,value2,days) {
+  if (days) {
+    var date = new Date();
+    date.setTime(date.getTime()+(days*24*60*60*1000));
+    var expires = "; expires="+date.toGMTString();
+  }
+  else var expires = "";
+  document.cookie = name+"="+value1+"~~"+value2+expires+"; path=/";
 };
 
 function readCookie(name) {
-    var nameEQ = name + "=";
-    var ca = document.cookie.split(';');
-    for(var i=0;i < ca.length;i++) {
-        var c = ca[i];
-        while (c.charAt(0)==' ') c = c.substring(1,c.length);
-        if (c.indexOf(nameEQ) == 0) return c.substring(nameEQ.length,c.length);
-    }
-    return null;
+  var nameEQ = name + "=";
+  var ca = document.cookie.split(';');
+  for(var i=0;i < ca.length;i++) {
+    var c = ca[i];
+    while (c.charAt(0)==' ') c = c.substring(1,c.length);
+    if (c.indexOf(nameEQ) == 0) return c.substring(nameEQ.length,c.length);
+  }
+  return null;
 };
 
 function eraseCookie(name) {
     createCookie(name,"",-1);
 };
+
+function getDefaultDateRange() {
+  var today = new Date();
+  var dd = today.getDate();
+  var mm = today.getMonth() + 1; //January is 0!
+  var yyyy = today.getFullYear();
+
+  if(dd < 10) {
+    dd = '0' + dd
+  } 
+
+  if(mm < 10) {
+    mm = '0' + mm
+  } 
+
+  today = mm + '/' + dd + '/' + yyyy;
+  twoWeeksAgo = new Date(today);
+  twoWeeksAgo.setDate(twoWeeksAgo.getDate() - 14);
+  return twoWeeksAgo + '~~' + today;
```

Comment:
Actually, furthermore, when twoWeeksAgo is made, it is in the form `Wed Jun 10 2015 00:00:00 GMT-0500 (CDT)` already; it's the other date (today) that is in the form `mm/dd/yyyy`. And the only reason it's like that is because when we gathered the different pieces of the date, all of them were separate. But the datepicker can figure out the dates from either, so unless one really cares about what the exact cookie is, it doesn't matter.